Library/Homebrew/cmd/tap.rb

```diff
@@ -39,59 +38,23 @@ def install_tap user, repo
     true
   end
 
-  def link_tap_formula(paths, warn_about_conflicts=true)
-    ignores = (HOMEBREW_LIBRARY/"Formula/.gitignore").read.split rescue []
-    tapped = 0
-
-    paths.each do |path|
-      to = HOMEBREW_LIBRARY.join("Formula", path.basename)
-
-      # Unexpected, but possible, lets proceed as if nothing happened
-      to.delete if to.symlink? && to.resolved_path == path
-
-      begin
-        to.make_relative_symlink(path)
-      rescue SystemCallError
-        to = to.resolved_path if to.symlink?
-        opoo <<-EOS.undent if warn_about_conflicts
-          Could not create link for #{Tty.white}#{tap_ref(path)}#{Tty.reset}, as it
-          conflicts with #{Tty.white}#{tap_ref(to)}#{Tty.reset}. You will need to use the
-          fully-qualified name when referring this formula, e.g.
-            brew install #{tap_ref(path)}
-          EOS
-      else
-        ignores << path.basename.to_s
-        tapped += 1
-      end
+  def migrate_taps(options={:autoskip => true})
+    # migrate taps from symlink based to directories based.
+    ignore = HOMEBREW_LIBRARY/"Formula/.gitignore"
+    if ignore.exist?
+      File.delete ignore
+    else
+      return if options[:autoskip]
     end
-
-    HOMEBREW_LIBRARY.join("Formula/.gitignore").atomic_write(ignores.uniq.join("\n"))
-
-    tapped
-  end
-
-  def repair_taps(warn_about_conflicts=true)
-    count = 0
-    # prune dead symlinks in Formula
     Dir.glob("#{HOMEBREW_LIBRARY}/Formula/*.rb") do |fn|
-      if not File.exist? fn
-        File.delete fn
-        count += 1
+      f = Pathname.new(fn)
+      next unless f.symlink?
+      if f.exist?
+        File.delete ignore if f.realpath.to_s.start_with? "#{HOMEBREW_LIBRARY}/Taps"
```

Comment:
`realpath` is something I've been trying to remove from the codebase where it isn't needed. There should only be one symlink, right? Then we can use `resolved_path`.